app/src/main/java/com/cornellappdev/transit/models/Payload.kt

```diff
@@ -0,0 +1,21 @@
+package com.cornellappdev.transit.models
+
+import com.squareup.moshi.Json
+
+/**
+ * Wrapper class to flatten Transit backend calls
+ */
+data class Payload<T>(
+    @Json(name = "success") var success: Boolean,
+    @Json(name = "data") var data: T
+) {
+    /**
+     * Unwrap the [data] in the response object. Fail-fast check on [success]
+     */
+    fun unwrap() : T {
+        if (!success){
+            throw Exception("Response unwrap failed: Success is false")
+        }
+        return data
+    }
```

Comment:
Hmm interesting, so we have this `unwrap()` function that will instantly throw an error if it's false? Is this the weird `success` field you showed me at Rilla? I presume it's fair to assume then that `success` is always true, hence the Exception.